From 06ba0d68e4d373734592f7d6028ef0008dddf155 Mon Sep 17 00:00:00 2001
From: Richard Zhao <b20223@freescale.com>
Date: Thu, 5 Mar 2009 17:43:11 +0800
Subject: [PATCH] ENGR00108905 sgtl5000: remove unneeded check of spk_amp_event

Remove unneeded check to make it more portable.

Signed-off-by: Richard Zhao <b20223@freescale.com>
(cherry picked from commit 08ba99785430aa69a24f5e2aef3ff59cb56c240e)
---
 sound/soc/imx/imx-3stack-sgtl5000.c |   22 +++++-----------------
 1 files changed, 5 insertions(+), 17 deletions(-)

diff --git a/sound/soc/imx/imx-3stack-sgtl5000.c b/sound/soc/imx/imx-3stack-sgtl5000.c
index dfdc2c9..aee6b5c 100644
--- a/sound/soc/imx/imx-3stack-sgtl5000.c
+++ b/sound/soc/imx/imx-3stack-sgtl5000.c
@@ -379,29 +379,17 @@ static int sgtl5000_set_spk(struct snd_kcontrol *kcontrol,
 
 static int spk_amp_event(struct snd_soc_dapm_widget *w,
 			 struct snd_kcontrol *kcontrol, int event) {
-	int hp_status;
 	struct mxc_sgtl5000_platform_data *plat;
-	struct imx_3stack_priv *priv;
-	priv = sgtl5000_3stack_pcm_link->machine->platform_data;
 	plat = sgtl5000_3stack_pcm_link->machine->pdev->dev.platform_data;
 
-	if ((plat->amp_enable == NULL) || (priv->reg_vdda == NULL))
+	if (plat->amp_enable == NULL)
 		return 0;
 
-	hp_status = plat->hp_status();
-	if (SND_SOC_DAPM_EVENT_ON(event)) {
-		if (hp_status) {
-			plat->amp_enable(0);
-			regulator_set_mode(priv->reg_vdda,
-					REGULATOR_MODE_NORMAL);
-		} else {
-			plat->amp_enable(1);
-			regulator_set_mode(priv->reg_vdda, REGULATOR_MODE_FAST);
-		}
-	} else {
+	if (SND_SOC_DAPM_EVENT_ON(event))
+		plat->amp_enable(1);
+	else
 		plat->amp_enable(0);
-		regulator_set_mode(priv->reg_vdda, REGULATOR_MODE_NORMAL);
-	}
+
 	return 0;
 }
 
-- 
1.5.4.4

